screens/AppNavigator/screens/Dex/DexAddLiquidity.tsx

```diff
@@ -0,0 +1,208 @@
+import { DexParamList } from './DexNavigator'
+
+type Props = StackScreenProps<DexParamList, 'AddLiquidity'>
+type EditingAmount = 'primary' | 'secondary'
+
+interface ExtPoolPairData extends PoolPairData {
+  aSymbol: string
+  bSymbol: string
+  aToBRate: BigNumber
+  bToARate: BigNumber
+}
+
+export function AddLiquidityScreen (props: Props): JSX.Element {
+  // TODO: poll for PoolPairData periodically
+
+  // this component state
+  const [tokenAAmount, setTokenAAmount] = useState<BigNumber>(new BigNumber(0))
+  const [tokenBAmount, setTokenBAmount] = useState<BigNumber>(new BigNumber(0))
+  const [sharePercentage, setSharePercentage] = useState<BigNumber>(new BigNumber(0))
+
+  // gather required data
+  const tokens = useTokensAPI()
+  const { pair: poolPairData } = props.route.params
+  const [aSymbol, bSymbol] = poolPairData.symbol.split('-')
+  const pair: ExtPoolPairData = {
+    ...poolPairData,
+    aSymbol,
+    bSymbol,
+    aToBRate: new BigNumber(poolPairData.tokenB.reserve).div(poolPairData.tokenA.reserve),
+    bToARate: new BigNumber(poolPairData.tokenB.reserve).div(poolPairData.tokenA.reserve)
+  }
+  const balanceA = tokens.find(at => at.id === pair.tokenA.id)
+  const balanceB = tokens.find(at => at.id === pair.tokenB.id)
+  const [tokenASymbol, tokenBSymbol] = pair.symbol.split('-')
+
+  const buildSummary = useCallback((ref: EditingAmount, refAmount: BigNumber) => {
+    if (ref === 'primary') {
+      setTokenAAmount(refAmount)
+      setTokenBAmount(refAmount.times(pair.aToBRate))
+      setSharePercentage(refAmount.div(pair.tokenA.reserve))
+    } else {
+      setTokenBAmount(refAmount)
+      setTokenAAmount(refAmount.times(pair.bToARate))
+      setSharePercentage(refAmount.div(pair.tokenB.reserve))
+    }
+  }, [tokenAAmount, tokenBAmount])
+
+  const grayDivider = <View style={tailwind('bg-gray w-full h-4')} />
+  return (
+    <View style={tailwind('w-full h-full')}>
+      <ScrollView style={tailwind('w-full flex-column flex-1')}>
+        {grayDivider}
+        {TokenInput(tokenASymbol, balanceA, tokenAAmount, (amount) => { buildSummary('primary', amount) })}
+        {grayDivider}
+        {TokenInput(tokenBSymbol, balanceB, tokenBAmount, (amount) => { buildSummary('secondary', amount) })}
+        {grayDivider}
+        {Summary(pair, sharePercentage)}
+      </ScrollView>
+      <View style={tailwind('w-full h-16')}>
+        {ContinueButton(
+          canAddLiquidity(
+            pair,
+            tokenAAmount,
+            tokenBAmount,
+            balanceA === undefined ? undefined : new BigNumber(balanceA.amount),
+            balanceB === undefined ? undefined : new BigNumber(balanceB.amount)
+          )
+        )}
+      </View>
+    </View>
+  )
+}
+
+function TokenInput (symbol: string, token: AddressToken | undefined, current: BigNumber, setState: (amount: BigNumber) => void): JSX.Element {
+  const renderIcon = (): JSX.Element|null => {
+    if (symbol === undefined) {
```

Comment:
You can remove this validation, we have a default fallback for icons since not all have icons